### ISSUE TITLE
Problem with template Haskell + Stack

### ISSUE THREAD

**Issue by fredefox** [ndmitchell/ghcid#192]
I have a stack-based project where I use `file-embed` (which uses template Haskell for embedding files into the compiled binary). This works fine when invoke `stack ghci` directly, however if I run `ghcid` on the project I get the following error:

```
<no location info>: error:
    
ByteCodeLink.lookupCE
During interactive linking, GHCi couldn't find the following symbol:
  MusteziGrammarziEmbed_grammar_closure
```

Just after checking the module with the template Haskell call to embed file. Any idea what could be causing this? Ideas for workarounds? The command line that ghcid reports to be invoking is

```
stack ghci --test --ghci-options=-fno-code --ghci-options=-fno-break-on-exception --ghci-options=-fno-break-on-error --ghci-options=-v1 --ghci-options=-ferror-spans --ghci-options=-j ...
```

So it's surprising to me that it's not doing the same.

**Comment by fredefox** [comment]
I think I see it, it's just being caused by the "-fno-code" flag.

**Comment by ndmitchell** [comment]
See https://github.com/ndmitchell/ghcid#faq, in particular:

> * _I get "During interactive linking, GHCi couldn't find the following symbol"._ This problem is a manifestation of [GHC bug 8025](https://ghc.haskell.org/trac/ghc/ticket/8025), which is fixed in GHC 8.4 and above. Ghcid automatically appends `-fno-code` to the command line, but for older GHC's you can supress that with `--test "return ()"` (to add a fake test) or `-c "ghci ..."` to manually specify the command to run.

I believe this is fixed in GHC 8.4.
